### ISSUE TITLE
Modify Classification {Voters, Deciders} to follow scikit-learn {ClassifierMixin, BaseEstimator} API's

### ISSUE THREAD

**Issue by levinwil** [neurodata/progressive-learning#226]
Guide on how to comply a class with a scikit-learn API: https://sklearn-template.readthedocs.io/en/latest/user_guide.html

ClassifierMixin docs: https://scikit-learn.org/stable/modules/generated/sklearn.base.ClassifierMixin.html#sklearn.base.ClassifierMixin

BaseEstimator docs: https://scikit-learn.org/stable/modules/generated/sklearn.base.BaseEstimator.html#sklearn.base.BaseEstimator

**BaseEstimator**
The following classes should inherit from BaseEstimator explicitly in their class names:
1. BaseVoter in base.py
2. BaseDecider in base.py

For example, you should be changing the class name of 'BaseVoter' to 'BaseVoter(BaseEstimator)'. 

**ClassifierMixin**
The following classes should inherit from ClassifierMixin explicitly in their class names:
1. TreeClassificationVoter in voter.py
2. KNNClassificationVoter in voter.py
3. BaseClassificationDecider in decider.py

For example, you should be changing the class name of 'TreeClassificationVoter(BaseVoter)' to 'TreeClassificationVoter(BaseVoter, BaseEstimator, ClassifierMixin)' 

The two important functions for ClassifierMixin are fit, predict, and predict_proba. 
1. fit: All of these classes already have fit methods, so you don't need to worry about that. 
2. predict_proba: BaseClassificationDecider (and its subclasses) already have predict and predict_proba, so you should just need to add 'ClassifierMixin' as a superclass in the BaseClassificationDecider class name. The voter predict_proba function is called "vote". So you'll have to change the name of the method from "vote" to "predict_proba".
3. predict: The voter currently doesn't have a "predict" function. So you'll have to make one. I would simply call the 'predict_proba' method (what is currently called 'vote') and return the argmax of each index. This is as simple as a one-line 'return np.argmax(self.predict_proba(X), axis = 1)'

**IMPORTANT NOTES**
Please please please please remember to reflect these changes in the docstrings of every class that you change. Any time you change a function name, please reflect that in the docstring for the class to which that function belonged. Any time you subclass from something, please reflect that in the docstring for that class.

You will also have to modify the ProgressiveLearner class in progressive_learner.py to reflect function name changes, namely from 'voter.vote' to 'voter.predict_proba'.

**Comment by emilyachang** [comment]
Could I be added onto this issue for Sprint 1? @jdey4
